kuma/users/views.py

```diff
@@ -138,27 +147,192 @@ def ban_user_and_cleanup_summary(request, user_id):
+    )
+    revisions_already_spam = [rev for rev in revisions_already_spam]
+    revisions_already_spam_by_distinct_doc = revision_by_distinct_doc(revisions_already_spam)
+
+    revisions_already_spam_ids = [rev.id for rev in revisions_already_spam]
+
+    not_identified_as_spam = revisions_from_last_three_days.exclude(
+        id__in=revisions_already_spam_ids).exclude(
+        id__in=revisions_to_mark_as_spam_and_revert.values_list('id', flat=True))
+    not_identified_as_spam = [rev for rev in not_identified_as_spam]
+    not_identified_as_spam_by_distinct_doc = revision_by_distinct_doc(not_identified_as_spam)
+
+    no_actions_taken = {
+        'revisions_already_identified_as_spam': revisions_already_spam_by_distinct_doc,
+        'revisions_not_identified_as_spam': not_identified_as_spam_by_distinct_doc
+    }
 
     context = {'detail_user': user,
                'form': UserBanForm(),
-               'revisions_reverted': revisions_reverted,
-               'revisions_needing_follow_up': revisions_needing_follow_up}
+               'actions_taken': actions_taken,
+               'needs_follow_up': needs_follow_up,
+               'no_actions_taken': no_actions_taken
+               }
 
     return render(request,
                   'users/ban_user_and_cleanup_summary.html',
                   context)
 
 
+def revision_by_distinct_doc(list_of_revisions):
+    if len(list_of_revisions):
+        list_of_revisions_distinct_doc = list(
+            Document.admin_objects.filter(
+                revisions__in=list_of_revisions
+            ).annotate(
+                latest_rev=Max('revisions')
+            ).values_list('latest_rev', flat=True)
+        )
+        list_of_revisions_distinct_doc = [rev for rev in list_of_revisions if rev.id in list_of_revisions_distinct_doc]
+    else:
+        list_of_revisions_distinct_doc = []
+
+    return list_of_revisions_distinct_doc
+
+
+@permission_required('users.add_userban')
+def revert_document(request, document_path, revision_id):
+    """
+    Revert document to a specific revision.
+    """
+    _, document_slug, __ = (
+        locale_and_slug_from_path(document_path, request))
+
+    revision = get_object_or_404(Revision.objects.select_related('document'),
+                                 pk=revision_id,
+                                 document__slug=document_slug)
+
+    comment = "spam"
+    document = revision.document
+    old_revision_pk = revision.pk
+    try:
+        new_revision = document.revert(revision, request.user, comment)
+        # schedule a rendering of the new revision if it really was saved
+        if new_revision.pk != old_revision_pk:
+            document.schedule_rendering('max-age=0')
+    except IntegrityError:
+        return False
+    return True
+
+
+@permission_required('wiki.delete_document')
+def delete_document(request, document_slug, document_locale):
+    """
+    Delete a Document.
+    """
```

Comment:
Similar to the comment on ``revert_document`` - the interface should be ``revision.document``, not the ``document``'s slug.